flink-core/src/main/java/org/apache/flink/core/fs/FSDataBufferedInputStream.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.fs;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * FSDataInputStream with Buffer, only support single thread.
+ * <p>
+ * The design of FSDataBufferedInputStream refers to {@link java.io.BufferedInputStream}.
+ * Compared to {@link java.io.BufferedInputStream}, the seek and getPos interfaces are mainly added.
+ */
+public class FSDataBufferedInputStream extends FSDataInputStream {
+
+	private static final int DEFAULT_BUFFER_SIZE = 8192;
+
+	protected byte[] buf;
+
+	// read offset of buf
+	private int pos;
+
+	// availed count of buf
+	private int count;
+
+	private final FSDataInputStream inputStream;
+
+	private boolean closed;
+
+	public FSDataBufferedInputStream(FSDataInputStream inputStream) {
+		this(inputStream, DEFAULT_BUFFER_SIZE);
+	}
+
```

Comment:
This constructor is only used in tests, I'd remove it here and add the constant to the test.